src/main/java/org/opentripplanner/middleware/controllers/api/TripHistoryController.java

```diff
@@ -41,11 +48,58 @@
     private static final String LIMIT_PARAM_NAME = "limit";
     private static final int DEFAULT_LIMIT = 10;
 
+    private final String ROOT_ROUTE;
+
+    public TripHistoryController(String apiPrefix) {
+        this.ROOT_ROUTE = apiPrefix + "secure/triprequests";
+    }
+
+    /**
+     * Register the API endpoint and GET resource to get trip requests
+     * when spark-swagger calls this function with the target API instance.
+     */
+    @Override
+    public void bind(final SparkSwagger restApi) {
+        ApiEndpoint apiEndpoint = restApi.endpoint(
+            endpointPath(ROOT_ROUTE).withDescription("Interface for retrieving trip requests."),
+            (q, a) -> LOG.info("Received request for 'triprequests' Rest API")
+        );
+        apiEndpoint
+            .get(path(ROOT_ROUTE)
+                    .withDescription("Gets a list of all trip requests for a user.")
+                    .withQueryParam()
+                        .withName("userId")
+                        .withRequired(true)
+                        .withDescription("The OTP user for which to retrieve trip requests.").and()
+                    .withQueryParam()
+                        .withName(LIMIT_PARAM_NAME)
+                        .withDefaultValue(String.valueOf(DEFAULT_LIMIT))
+                        .withDescription("If specified, the maximum number of trip requests to return, starting from the most recent.").and()
+                    .withQueryParam()
+                        .withName(FROM_DATE_PARAM_NAME)
+                        .withPattern("yyyy-MM-dd")
+                        .withDefaultValue("The current date")
+                        .withDescription("If specified, the earliest date (format yyyy-MM-dd) for which trip requests are retrieved.").and()
+                    .withQueryParam()
+                        .withName(TO_DATE_PARAM_NAME)
+                        .withPattern("yyyy-MM-dd")
```

Comment:
Let's make any date or time string template string something that is some global constant so that we never have any differences in date formats.

Reply:
Addressed in https://github.com/ibi-group/otp-middleware/pull/55/commits/6f19f22cccf1abe83b5d151ef4aa7888c711f726.